pkg/endpoint/bpf.go

```diff
@@ -1228,13 +1269,44 @@ func (e *Endpoint) applyPolicyMapChanges() error {
 	return nil
 }
 
+// endpointPolicyLockdown puts the endpoint policy map into a lockdown
+// mode. A new bpf policy map is populated with deny all traffic entries,
+// the old map is unpinned, and the new map is pinned.
+func (e *Endpoint) endpointPolicyLockdown() error {
+	tmpPMPath := e.policyMapPath() + "_lockdown"
+	pM, err := policymap.CreateUnpinned(tmpPMPath)
+	if err != nil {
+		return fmt.Errorf("could not create bpf map %q: %w", tmpPMPath, err)
+	}
+	mapState := policy.NewMapState(nil)
+	for _, k := range allKeys {
+		policymapKey := policymap.NewKey(k.Identity, k.DestPort,
+			k.Nexthdr, k.TrafficDirection)
+		if err := pM.DenyKey(policymapKey); err != nil {
+			return fmt.Errorf("failed to add deny all policy: %w", err)
+		}
+		mapState.Insert(k, policy.MapStateEntry{IsDeny: true})
+	}
+	err = e.policyMap.Unpin()
+	if err != nil {
+		return fmt.Errorf("failed to unpin previous map: %w", err)
+	}
+	e.policyMap = pM
+	e.realizedPolicy.SetPolicyMap(mapState)
+	if err = e.policyMap.Pin(e.policyMapPath()); err != nil {
```

Comment:
There will be a time period in which no map is pinned to the given path. I guess that this should be fine, though, as AFAIU the datapath will just drop all traffic in this case. Is my understanding correct?

Reply:
That is correct, and the endpoint will regenerate, as policy sync will fail if this method fails.